supervisor/shared/usb/usb.c

```diff
@@ -24,79 +24,47 @@
  * THE SOFTWARE.
  */
 
-#include "nrfx.h"
-#include "nrfx_power.h"
 #include "tick.h"
-#include "usb.h"
+#include "shared-bindings/microcontroller/Processor.h"
+#include "supervisor/port.h"
+#include "supervisor/usb.h"
 #include "lib/utils/interrupt_char.h"
 #include "lib/mp-readline/readline.h"
 
-#ifdef SOFTDEVICE_PRESENT
-#include "nrf_sdm.h"
-#include "nrf_soc.h"
-#endif
-
-/* tinyusb function that handles power event (detected, ready, removed)
- * We must call it within SD's SOC event handler, or set it as power event handler if SD is not enabled.
- */
-extern void tusb_hal_nrf_power_event(uint32_t event);
-
-void usb_init(void) {
-
-    // USB power may already be ready at this time -> no event generated
-    // We need to invoke the handler based on the status initially
-    uint32_t usb_reg;
-
-#ifdef SOFTDEVICE_PRESENT
-    uint8_t sd_en = false;
-    (void) sd_softdevice_is_enabled(&sd_en);
-
-    if ( sd_en ) {
-        sd_power_usbdetected_enable(true);
-        sd_power_usbpwrrdy_enable(true);
-        sd_power_usbremoved_enable(true);
-
-        sd_power_usbregstatus_get(&usb_reg);
-    }else
-#endif
-    {
-        // Power module init
-        const nrfx_power_config_t pwr_cfg = { 0 };
-        nrfx_power_init(&pwr_cfg);
-
-        // Register tusb function as USB power handler
-        const nrfx_power_usbevt_config_t config = { .handler = (nrfx_power_usb_event_handler_t) tusb_hal_nrf_power_event };
-        nrfx_power_usbevt_init(&config);
-
-        nrfx_power_usbevt_enable();
-
-        usb_reg = NRF_POWER->USBREGSTATUS;
-    }
+#include "tusb.h"
 
-    if ( usb_reg & POWER_USBREGSTATUS_VBUSDETECT_Msk ) {
-        tusb_hal_nrf_power_event(NRFX_POWER_USB_EVT_DETECTED);
-    }
-
-    if ( usb_reg & POWER_USBREGSTATUS_OUTPUTRDY_Msk ) {
-        tusb_hal_nrf_power_event(NRFX_POWER_USB_EVT_READY);
-    }
+// Serial number as hex characters. This writes directly to the USB
+// descriptor.
+extern uint16_t usb_serial_number[1 + COMMON_HAL_MCU_PROCESSOR_UID_LENGTH * 2];
 
+void load_serial_number(void) {
     // create serial number based on device unique id
-    extern uint16_t usb_desc_str_serial[1 + 16];
+    uint8_t raw_id[COMMON_HAL_MCU_PROCESSOR_UID_LENGTH];
+    common_hal_mcu_processor_get_uid(raw_id);
+
+    const char nibble_to_hex[16] = {'0', '1', '2', '3', '4', '5', '6', '7', '8', '9',
+                                    'A', 'B', 'C', 'D', 'E', 'F'};
+    for (int i = 0; i < COMMON_HAL_MCU_PROCESSOR_UID_LENGTH; i++) {
+        for (int j = 0; j < 2; j++) {
```

Comment:
Old code inverted the order due to endianness. Did you just remove that because it's unique anyway?

Reply:
Ya, I don't care about the order.